langchain/client/langchain.py

```diff
@@ -136,11 +142,11 @@ def _headers(self) -> Dict[str, str]:
         """Get the headers for the API request."""
         headers = {}
         if self.api_key:
-            headers["authorization"] = f"Bearer {self.api_key}"
+            headers["x-api-key"] = self.api_key
         return headers
 
     @property
-    def query_params(self) -> Dict[str, str]:
+    def query_params(self) -> Dict[str, Any]:
         """Get the headers for the API request."""
         return {"tenant_id": self.tenant_id}
```

Comment:
Since this is optional now, should we do an `ensure_tenant_id` / re-call if not provided here?

Reply:
We fetch and throw in the validator if we can't get tenant_id so that pretty much ensures it much be set